### PR TITLE
javers api PoC

### PR DESCRIPTION
**Pull request by pszymczyk**

### CODE REVIEW

File: javers-api/src/main/java/org/javers/api/JaversQueryService.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        this.javers = javers;
+    }
+
+    public List<CdoSnapshot> findSnapshots(String instanceId, String className) {
+        JqlQuery jqlQuery = QueryBuilder.byInstanceId(instanceId, quietlyClassForName(className)).build();
+
+        return javers.findSnapshots(jqlQuery);
+    }
+
+    private Class<?> quietlyClassForName(String className) {
```
Comment (bartoszwalacik) [inline javers-api/src/main/java/org/javers/api/JaversQueryService.java]: already in ReflectionUtil.classForName()

File: javers-api/src/main/java/org/javers/api/SnapshotResponseMessageConverter.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ */
+class SnapshotResponseMessageConverter extends AbstractHttpMessageConverter<SnapshotsResponse> {
+
+    private final JsonConverter jsonConverter;
+
+    public SnapshotResponseMessageConverter(Javers javers) {
+        this(javers.getJsonConverter());
+    }
+
+    public SnapshotResponseMessageConverter(JsonConverter jsonConverter) {
```
Comment (bartoszwalacik) [inline javers-api/src/main/java/org/javers/api/SnapshotResponseMessageConverter.java]: protected?

File: javers-api/src/main/java/org/javers/api/SnapshotResponseMessageConverter.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        this.jsonConverter = jsonConverter;
+    }
+
+    @Override
+    protected boolean supports(Class<?> clazz) {
+        return SnapshotsResponse.class.equals(clazz);
+    }
+
+    @Override
+    protected SnapshotsResponse readInternal(Class<? extends SnapshotsResponse> clazz, HttpInputMessage inputMessage) throws IOException, HttpMessageNotReadableException {
```
Comment (bartoszwalacik) [inline javers-api/src/main/java/org/javers/api/SnapshotResponseMessageConverter.java]: do we need read at this point?

File: javers-api/src/main/java/org/javers/api/SnapshotsResponse.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * @author pawel szymczyk
+ */
+public class SnapshotsResponse {
+
+    private List<CdoSnapshot> snapshots;
+
+    protected SnapshotsResponse() {
+    }
```
Comment (bartoszwalacik) [inline javers-api/src/main/java/org/javers/api/SnapshotsResponse.java]: snapshots = Collections.emptyList()

File: javers-api/src/test/java/org/javers/api/JaversMongoStarterIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    @Autowired
+    private Javers javers;
+
+    @Before
+    public void setup() {
+        restTemplate.getRestTemplate().getMessageConverters().add(0, new SnapshotResponseMessageConverter(javers));
+    }
+
+    @Test
+    public void shouldGetSnapshotForGivenInstanceId() {
```
Comment (bartoszwalacik) [inline javers-api/src/test/java/org/javers/api/JaversMongoStarterIntegrationTest.java]: JUnit ... why?

File: javers-core/src/main/java/org/javers/repository/jql/QueryBuilder.java [hunk 6]
```diff
@@ -377,4 +377,5 @@ public QueryBuilder byAuthor(String author) {
     public JqlQuery build(){
         return new JqlQuery(filter, queryParamsBuilder.build(), shadowScope);
     }
+
```
Comment (bartoszwalacik) [inline javers-core/src/main/java/org/javers/repository/jql/QueryBuilder.java]: why?

File: javers-core/src/test/groovy/org/javers/repository/jql/QueryBuilderTest.groovy [hunk 7]
```diff
@@ -0,0 +1,20 @@
+package org.javers.repository.jql
+
+import spock.lang.Specification
+
+class QueryBuilderTest extends Specification {
```
Comment (bartoszwalacik) [inline javers-core/src/test/groovy/org/javers/repository/jql/QueryBuilderTest.groovy]: omg 👎

**Comment by pszymczyk** [comment]
I close until the delivery of the MVP

**Pull request closed without merging**